src/XRP/xrp-path-element.ts

```diff
@@ -0,0 +1,29 @@
+import { Payment } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import XRPCurrency from './xrp-currency'
+
+type PathElement = Payment.PathElement
+
+/*
+ * A path step in an XRP Ledger Path.
+ * @see: https://xrpl.org/paths.html#path-steps
+ */
+export default class XRPPathElement {
+  public static from(pathElement: PathElement): XRPPathElement | undefined {
+    const account = pathElement.getAccount()?.getAddress()
+    const currency = pathElement.getCurrency()
+    let xrpCurrency
+    if (currency) {
```

Comment:
if you drop `undefined` from the definition above, this if statement is removeable.

Reply:
I can't actually get rid of this if statement, because `currency` here comes from the protobuf and has type `Currency | undefined` based on how the protos are written. (it's not an XRPCurrency)